### PR TITLE
Fix issue #30

### PR DESCRIPTION
**Pull request by jnury**
Hello,

This is my fix proposal for issue #30.

Before registering the PSSessionConfiguration, we save the list of services running in the same process as WinRM + services depending from them.

We let Register-PSSessionConfiguration command 10 seconds to complete. If it not complete in 10 seconds and the WinRM service is in the state 'Stopping', we kill the process hosting WinRm service and then restart all services impacted.

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/jea/46)

### CODE REVIEW

**Comment by rpsqrd** [comment]
Thanks Julien -- I'll take a closer look this week and get this merged in. How would we suggest people opt out of this feature if they don't want the risk of killing the process? Set the timeout to `[int]::MaxValue` or `null`?

**Comment by jnury** [comment]
Hello Ryan,
You are right, we need an option to opt out. I think setting the timeout to
0 is the most intuitive way.
I’ll modify my PR the next weekend.
Have a nice day.

**Comment by jnury** [comment]
Disabling WinRM workaround (process killing) is implemented: when HungRegistrationTimeout is set to 0, no process is killed.

File: DSC Resource/JustEnoughAdministration/JustEnoughAdministration.psm1 [hunk 1]
```diff
@@ -611,13 +620,152 @@ class JeaEndpoint
...
 
+    ## Get a PS Session Configuration based on its name
+    hidden [Object] GetPSSessionConfiguration($Name)
+    {
+        $winRMService = Get-Service -Name 'WinRM'
+        if ($winRMService -and $winRMService.Status -eq 'Running')
+        {
+            # Temporary disabling Verbose as xxx-PSSessionConfiguration methods verbose messages are useless for DSC debugging
+            $VerbosePreferenceBackup = $Global:VerbosePreference
+            $Global:VerbosePreference = 'SilentlyContinue'
```
Comment (TravisEz13) [inline DSC Resource/JustEnoughAdministration/JustEnoughAdministration.psm1]: Can this be changed to the script scope?

File: DSC Resource/JustEnoughAdministration/JustEnoughAdministration.psm1 [hunk 2]
```diff
@@ -611,13 +620,152 @@ class JeaEndpoint
...
+
+    ## Unregister a PS Session Configuration based on its name
+    hidden [Void] UnregisterPSSessionConfiguration($Name)
+    {
+        $winRMService = Get-Service -Name 'WinRM'
+        if ($winRMService -and $winRMService.Status -eq 'Running')
+        {
+            # Temporary disabling Verbose as xxx-PSSessionConfiguration methods verbose messages are useless for DSC debugging
+            $VerbosePreferenceBackup = $Global:VerbosePreference
+            $Global:VerbosePreference = 'SilentlyContinue'
```
Comment (TravisEz13) [inline DSC Resource/JustEnoughAdministration/JustEnoughAdministration.psm1]: same scope issue

File: DSC Resource/JustEnoughAdministration/JustEnoughAdministration.psm1 [hunk 3]
```diff
@@ -611,13 +620,152 @@ class JeaEndpoint
...
+                    if ($dependentService.Status -eq 'Running' -and $serviceList -notcontains $dependentService.Name)
+                    {
+                        $serviceList += $dependentService.Name
+                    }
+                }
+            }
+
+            if ($Path)
+            {
+                $registerString = "`$null = Register-PSSessionConfiguration -Name '$Name' -Path '$Path' -Force -ErrorAction 'Stop' -WarningAction 'SilentlyContinue'"
```
Comment (TravisEz13) [inline DSC Resource/JustEnoughAdministration/JustEnoughAdministration.psm1]: Can we use a static script block with paramenters and the argument list to Start-job?

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
see comments

**Pull request closed without merging**